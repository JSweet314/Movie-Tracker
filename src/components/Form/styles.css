form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

input {
  width: 20%;
  margin: 1em 0 1em 2em;
  height: 1.5em;
  font-size: 1em;
  border: none;
  border-bottom: 1px solid #300305;
}

button[type='submit'] {
  width: 21%;
  height: 2.8em;
  margin: 1.2em 0 0 3em;
  background-color: #5D020B;
  color: white;
}
