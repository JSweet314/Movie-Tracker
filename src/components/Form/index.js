import React from 'react';
import './styles.css';
import PropTypes from 'prop-types';

const Form = (
  {errorMessage, name, email, password, routeId, handleChange, handleSubmit}
) => {
  const display = routeId === 'signup' ? "inline" : "none";
  const submitText = routeId === 'signup' ? "Sign Up" : "Log In";
  return (
    <form className='form' onSubmit={event => handleSubmit(event)}>
      <input
        style={{display}}
        type='text'
        name='name'
        placeholder="name"
        value={name}
        onChange={event => handleChange(event)}
      />
      <input
        type='email'
        name='email'
        placeholder='email'
        value={email}
        onChange={event => handleChange(event)}
      />
      <input
        type='password'
        name='password'
        placeholder='password'
        value={password}
        onChange={event => handleChange(event)}
      />
      <button type='submit'>{submitText}</button>
      <p>{errorMessage}</p>
    </form>
  );
};

Form.propTypes = {
  errorMessage: PropTypes.string,
  name: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  routeId: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

export default Form;
