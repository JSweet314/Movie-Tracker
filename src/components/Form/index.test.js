import React from 'react';
import { shallow } from 'enzyme';
import Form from './index';

describe('Form', () => {
  const mockHandleChange = jest.fn();
  const mockHandleSubmit = jest.fn();

  it('should match the snapshot on signup page', () => {
    const wrapper = shallow(
      <Form 
        name={''}
        email={''}
        password={''}
        routeId={'signup'}
        handleChange={mockHandleChange}
        handleSubmit={mockHandleSubmit}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot on login page', () => {
    const wrapper = shallow(
      <Form 
        name={''}
        email={''}
        password={''}
        routeId={'login'}
        handleChange={mockHandleChange}
        handleSubmit={mockHandleSubmit}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
