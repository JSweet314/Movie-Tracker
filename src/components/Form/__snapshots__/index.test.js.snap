// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot on login page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    email=""
    handleChange={[Function]}
    handleSubmit={[Function]}
    name=""
    password=""
    routeId="login"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="name"
          onChange={[Function]}
          placeholder="name"
          style={
                    Object {
                              "display": "none",
                            }
          }
          type="text"
          value=""
/>,
        <input
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
          value=""
/>,
        <input
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
/>,
        <button
          type="submit"
>
          Log In
</button>,
        <p />,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "name",
          "onChange": [Function],
          "placeholder": "name",
          "style": Object {
            "display": "none",
          },
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log In",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Log In",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="name"
            onChange={[Function]}
            placeholder="name"
            style={
                        Object {
                                    "display": "none",
                                  }
            }
            type="text"
            value=""
/>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
/>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
          <button
            type="submit"
>
            Log In
</button>,
          <p />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "placeholder": "name",
            "style": Object {
              "display": "none",
            },
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log In",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Log In",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Form should match the snapshot on signup page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    email=""
    handleChange={[Function]}
    handleSubmit={[Function]}
    name=""
    password=""
    routeId="signup"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="name"
          onChange={[Function]}
          placeholder="name"
          style={
                    Object {
                              "display": "inline",
                            }
          }
          type="text"
          value=""
/>,
        <input
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
          value=""
/>,
        <input
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
/>,
        <button
          type="submit"
>
          Sign Up
</button>,
        <p />,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "name",
          "onChange": [Function],
          "placeholder": "name",
          "style": Object {
            "display": "inline",
          },
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="name"
            onChange={[Function]}
            placeholder="name"
            style={
                        Object {
                                    "display": "inline",
                                  }
            }
            type="text"
            value=""
/>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
/>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
          <button
            type="submit"
>
            Sign Up
</button>,
          <p />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "placeholder": "name",
            "style": Object {
              "display": "inline",
            },
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
